modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/core/hooks/useDataView.es.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {DataDefinitionSchema} from '../schema/DataDefinition.es';
+import {DataLayoutSchema} from '../schema/DataLayout.es';
+import {SYMBOL_RAW} from '../schema/Schema.es';
+
+const SCHEMAS_DICTIONARY = {
+	dataDefinition: DataDefinitionSchema,
+	dataLayout: DataLayoutSchema,
+};
+
+/**
+ * Context stores all instances of schemas created by key to avoid recreating
+ * a new instance of the same Schema in different components.
+ */
+const context = {};
```

Comment:
Why not using a `new Map()` here?

Reply:
@diegonvs using `Map` here would be somewhat unnecessary for the final object of this implementation, we should consider using it only in some cases, such as:

- Key Order, if it matters. (We don't need this here)
- Size, in case we need to get the size of our map. (We don't need this here)
- Iteration, just in case we need to iteration items. (We don't need this here)
- If we have a constant addition and removal of key values of the object, if this is very intense we should certainly consider `Map` but this is also not the case here.

Therefore, going with just one object will be more "simple" is no problem for us at this moment.